crates/router/src/connector/stripe.rs

```diff
@@ -1732,15 +1732,23 @@ impl api::IncomingWebhook for Stripe {
             .change_context(errors::ConnectorError::WebhookReferenceIdNotFound)?;
 
         Ok(match details.event_data.event_object.object {
-            stripe::WebhookEventObjectType::PaymentIntent => {
-                api_models::webhooks::ObjectReferenceId::PaymentId(
-                    api_models::payments::PaymentIdType::ConnectorTransactionId(
-                        details.event_data.event_object.id,
+            stripe::WebhookEventObjectType::PaymentIntent
+            | stripe::WebhookEventObjectType::Charge => {
+                match details.event_data.event_object.metadata {
+                    // if order_id is present
+                    Some(meta_data) => api_models::webhooks::ObjectReferenceId::PaymentId(
+                        api_models::payments::PaymentIdType::PaymentAttemptId(meta_data.order_id),
                     ),
-                )
+                    // else used connector_transaction_id
+                    None => api_models::webhooks::ObjectReferenceId::PaymentId(
+                        api_models::payments::PaymentIdType::ConnectorTransactionId(
+                            details.event_data.event_object.id,
```

Comment:
In case of `stripe::WebhookEventObjectType::Charge`, event_object.id is charge_id instead of payment_id right? Do we need to consume charge webhooks for payment flows?

Reply:
Refactored